pkg/controller/vsphere/reconciler_test.go

```diff
@@ -463,6 +463,81 @@ func TestGetNetworkDevices(t *testing.T) {
 	}
 }
 
+func TestGetDiskSpec(t *testing.T) {
+	model, session, server := initSimulator(t)
+	defer model.Remove()
+	defer server.Close()
+
+	managedObj := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+	objVM := object.NewVirtualMachine(session.Client.Client, managedObj.Reference())
+
+	testCases := []struct {
+		name          string
+		expectedError error
+		devices       func() object.VirtualDeviceList
+	}{
+		{
+			name: "Succefully get disk spec",
+			devices: func() object.VirtualDeviceList {
+				devices, err := objVM.Device(context.TODO())
+				if err != nil {
+					t.Fatal(err)
+				}
+				return devices
+			},
+		},
+		{
+			name: "Fail on invalid disk count",
+			devices: func() object.VirtualDeviceList {
+				devices, err := objVM.Device(context.TODO())
+				if err != nil {
+					t.Fatal(err)
+				}
+				devices = append(devices, &types.VirtualDisk{})
+				return devices
+			},
+			expectedError: errors.New("invalid disk count: 2"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			machineScope := &machineScope{
+				Context: context.TODO(),
+				providerSpec: &vsphereapi.VSphereMachineProviderSpec{
+					DiskGiB: 1,
```

Comment:
Could possibly add this field to the `testCases` and try a couple of different values to make this test suite a little more thorough? WDYT?

Reply:
I'm not getting this, you suggest trying different disk sizes?

Reply:
Yeah, the disk size (and expected disk size) could come from the test case and then you could try 1, 2, 3 as values in different cases and check the calculation of the size is correct, WDYT?

Reply:
make sense, more tests never hurt :D

Reply:
@JoelSpeed done